### ISSUE TITLE
Autokey main window open at startup; "Show main window at startup" is unchecked.

### ISSUE THREAD

**Issue by mapsedge** [autokey/autokey#928]
### AutoKey is a Xorg application and will not function in a Wayland session. Do you use Xorg (X11) or Wayland?

Xorg

### Has this issue already been reported?

- [X] I have searched through the existing issues.

### Is this a question rather than an issue?

- [X] This is not a question.

### What type of issue is this?

UI/Usability

### Choose one or more terms that describe this issue:

- [ ] autokey triggers
- [ ] autokey-gtk
- [X] autokey-qt
- [ ] beta
- [ ] bug
- [ ] critical
- [ ] development
- [ ] documentation
- [ ] enhancement
- [ ] installation/configuration
- [ ] phrase expansion
- [ ] scripting
- [ ] technical debt
- [X] user interface

### Which Linux distribution did you use?

Kubuntu 22.04, Plasma 5.24.7, although this has been an issue I've noticed since Kubuntu 20

### Which AutoKey GUI did you use?

Qt

### Which AutoKey version did you use?

0.96

### How did you install AutoKey?

.deb package from github

### Can you briefly describe the issue?

Autokey main window open at startup; "Show main window at startup" is unchecked.

### Can the issue be reproduced?

None

### What are the steps to reproduce the issue?

Restart PC. Log in. Hellooooooo Autokey!

### What should have happened?

Autokey should have started in the background without displaying the interface.

### What actually happened?

Autokey displayed the interface at startup.

**Comment by josephj11** [comment]
I'm assuming this only happens when you log in to your user account, not when you close the AutoKey application (not just the menu window) and then restart it.

I'm running Kubuntu 22.04 and do not have this problem.

We have to find the command that is starting AutoKey when you log in.

Take a look in $HOME/.config/autostart and in $HOME/.config/autostart-scripts and see if there's a script that invokes AutoKey. If there is and the command has the -c or --configure flag specified, remove that flag.

If you have the AutoKey menu window open when you log out, KDE might see it and add it to its session restore information if you have your DTE set to restore the previous session upon login. 

I don't immediately see where that setting is stored. If the above doesn't fix it, this will require some more research.

**Comment by mapsedge** [comment]
You are correct, this is only when logging into my user account. My desktop starts fresh, no restoring of the session.

There is an autokey.desktop in autostart, but the exec line doesn't have the configure flag.

**Comment by josephj11** [comment]
The only other workaround I can think of would be to remove/rename that .desktop file and add your own startup script for AutoKey and see if that helps.

First, just remove/rename it just in case AutoKey is getting started by something somewhere else. If AutoKey still starts, we'll have to find out what's doing it.

**Comment by mjeveritt** [comment]
Same issue here, and whilst I wouldn't ordinarily 'bump' threads, I have subscribed, and I will do a bit more further testing of my environment...
'problem' system here is `PRETTY_NAME="Ubuntu 22.04.3 LTS"`
desktop: `ii  plasma-workspace                4:5.24.7-0ubuntu0.1`

**Comment by josephj11** [comment]
@mjeveritt This could be caused by AutoKey starting twice: once from the `Automatically start AutoKey after login` setting in AutoKey and once from KDE session restore. Try turning off that option and see what happens.

**Comment by mjeveritt** [comment]
> [@mjeveritt](https://github.com/mjeveritt) This could be caused by AutoKey starting twice: once from the `Automatically start AutoKey after login` setting in AutoKey and once from KDE session restore. Try turning off that option and see what happens.

Funny you should mention that .. and it wouldn't be the first time .. and Yes, I had thought that often a re-loading of an application on logon/startup will bring the window to the foreground .. thanks!

**Comment by josephj11** [comment]
@mjeveritt  It's a feature, not  bug. :) It doesn't bring the window to the foreground.

When you launch AutoKey without the `-c` or `--configure` options, it starts in the background with no visible window. This is the default because most of the time AutoKey doesn't need configuration, it just needs to work. If you start AutoKey a second time, it detects that it is already running and displays the main window then exits, leaving the first instance running.

**Comment by mjeveritt** [comment]
Having learnt of the existence of `~/.config/ksmserverrc` I can now verify what apps are starting themselves via the XDG methods, and via KDE's session management, as above 🥳 .

This has allowed me to fix my issue here, [thanks!] and also troubleshot a hiccup I was having with Barrier not auto-connecting properly, and showing Its window! 👍 

I won't even mention the learning I just got from ChatGPT about using python to interface with pulseaudio just! 🤯

**Comment by josephj11** [comment]
@mjeveritt Glad it's sorted.

I gave up trying to figure out pulseaudio years ago and I was lurking on their listserve! That stuff needs AI to untangle it!

Never heard of Barrier before. It looks interesting.
